view/frontend/web/js/view/payment/method-renderer/omise-offsite-installment-method.js

```diff
@@ -102,14 +205,36 @@ define(
 
             /**
              * Returns Installment Terms
-             *
+             * @param {string} id - Bank id
              * @return {array}
              */
-            getInstallmentTerms(id) {
+            getInstallmentTerms: function (id) {
                 const installmentBackends = checkoutConfig.installment_backends;
+                const templateLabel = $.mage.__('%terms months (%amount / month)');
+
                 for (const key in installmentBackends) {
-                    if (installmentBackends[key]._id === 'installment_' + id)
-                        return ko.observableArray(installmentBackends[key].allowed_installment_terms);
+                    if (installmentBackends[key]._id !== 'installment_' + id) {
+                        continue;
+                    }
+
+                    let dispTerms = [];
+                    const terms = installmentBackends[key].allowed_installment_terms;
+                    const minSingleInstallment = this.getInstallmentMinimum(id);
+
+                    for (let i = 0; i < terms.length; i++) {
+                        const amount = this.calculateSingleInstallmentAmount(id, terms[i]);
+
+                        if (amount >= minSingleInstallment) {
+                            dispTerms.push({
+                                label: templateLabel.replace('%terms', terms[i]).replace('%amount', this.getFormattedAmount(amount)),
+                                key: terms[i]
+                            });
+                        }
+                    }
```

Comment:
What if we add `break;` here to preventing an unnecessary iteration?
(i.e. if term 8 months is lower than `minSingleInstallment` then, stop it from calculating for 9 months, 10 months, 12 months, and so on)
```js
for (let i = 0; i < terms.length; i++) {
    const amount = this.calculateSingleInstallmentAmount(id, terms[i]);

    if (amount < minSingleInstallment) {
        break;
    }

    dispTerms.push({
        label: templateLabel.replace('%terms', terms[i]).replace('%amount', this.getFormattedAmount(amount)),
        key: terms[i]
    });
}
```

Reply:
what guarantee we have that it will be this way sorted every time when coming from backend?